src/main/java/seedu/address/model/Model.java

```diff
@@ -91,6 +99,38 @@ public interface Model {
      */
     void resetPersons();
 
+    /**
+     * Returns true if a recommendation with the same identity as
+     * {@code recommendation} exists in the address book.
+     */
+    boolean hasRecommendation(Recommendation recommendation);
+
+    /**
+     * Deletes the given recommendation.
+     * The recommendation must exist in the address book.
+     */
+    void deleteRecommendation(Recommendation target);
+
+    /**
+     * Adds the given recommendation.
+     * {@code recommendation} must not already exist in the address book.
+     */
+    Recommendation addRecommendation(Recommendation recommendation);
+
+    /**
+     * Replaces the given recommendation {@code target} with {@code editedRecommendation}.
+     * {@code target} must exist in the address book.
+     * The recommendation identity of {@code editedRecommendation}
+     * must not be the same as another existing recommendation in the address book.
+     */
+    void setRecommendation(Recommendation target, Recommendation editedRecommendation);
+
+    /**
+     * Clears the person list.
+     * Used for sorting the list, as the list needs to be cleared before it can be sorted.
+     */
+    void resetRecommendations();
```

Comment:
Most of these methods are probably pointless. But I didn't want to risk it so I copied the functionality of PersonList one-for-one.